avocado/utils/vmimage.py

```diff
@@ -0,0 +1,375 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2017
+# Author: Amador Pahim <apahim@redhat.com>
+
+"""
+Provides VM images acquired from official repositories
+"""
+
+import os
+import re
+import string
+import tempfile
+import urllib2
+import uuid
+
+try:
+    import lzma
+    LZMA_CAPABLE = True
+except ImportError:
+    LZMA_CAPABLE = False
+
+from HTMLParser import HTMLParser
+
+from . import asset
+from . import path as utils_path
+from . import process
+
+
+class ImageProviderError(Exception):
+    """
+    Generic error class for ImageProvider
+    """
+
+
+class VMImageHtmlParser(HTMLParser):
+    def __init__(self, pattern):
+        HTMLParser.__init__(self)
+        self.versions = []
+        self.pattern = pattern
+
+    def handle_starttag(self, tag, attrs):
+        if tag != 'a':
+            return
+        for attr in attrs:
+            if attr[0] == 'href' and re.match(self.pattern, attr[1]):
+                self.versions.append(attr[1].strip('/'))
+
+
+class ImageBase(object):
+    def __init__(self, **kwargs):
+        self.path = None
+        self.url = None
+
+        self.url_versions = None
+        self.url_images = None
+        self.image = None
+
+        self.default_version = None
+        self.default_build = None
+        self.default_arch = None
+
+        self.args = {}
+        self.initargs = kwargs
+
+    def get(self):
+        self.url = self.get_url()
+
+        asset_path = asset.Asset(name=self.url,
+                                 asset_hash=self.args['checksum'],
+                                 algorithm=self.args['algorithm'],
+                                 locations=None,
+                                 cache_dirs=[self.args['cache_dir']],
+                                 expire=None).fetch()
+
+        if os.path.splitext(asset_path)[1] == '.xz':
+            if LZMA_CAPABLE:
+                asset_path = self._extract(asset_path)
```

Comment:
Well I don't think you should treat this as `_extract` would raise the right exception with details about `xz`...

Reply:
OK, I see a reason for the LZMA_CAPABLE, but not here, but in `CentOSImageProvider`, where based on availability of LZMA you should modify the `self.image`.

Reply:
Ack. Moved to the CentOSImageProvider.